lib/formatters/json.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * @fileoverview JSON reporter
+ * @author Burak Yigit Kaya aka BYK
+ * @copyright 2015 Burak Yigit Kaya. All rights reserved.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Public Interface
+//------------------------------------------------------------------------------
+
+module.exports = function(results) {
+    var output = {};
+
+    results.forEach(function(result) {
+        output[result.filePath] = (output[result.filePath] || []).concat(result.messages);
+    });
+
+    return JSON.stringify(output);
```

Comment:
@nzakas - not sure what you're referring to. All I'm doing is grouping the results by file name so you get something like:
```
{
    "foo.js": [ ... ],
    "bar.js": [ ... ]
}
```

where the contents of these arrays are unmodified `result` objects.